### PR TITLE
Avoid running Calico datastore migration pods when Calico is in policy only mode

### PR DESCRIPTION
**Pull request by whites11**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: files/conf/k8s-addons [hunk 1]
```diff
@@ -88,13 +88,15 @@ echo "kube-proxy successfully installed"
 # restart ds to apply config from configmap
 kubectl delete pods -l app.kubernetes.io/name=kube-proxy -n kube-system
 
+{{- if not .CalicoPolicyOnly }}
 # TODO: remove migrator once all clusters have been migrated to kubernetes datastore
 kubectl apply -f /srv/calico-crd-installer-rbac.yaml
 for job in calico-crd-installer calico-datastore-migrator-pre
```
Comment (tfussell) [inline files/conf/k8s-addons:94]: We need `calico-crd-installer` for policy-only and full Calico so we'll have to drop the `for` loop and split this in two parts.

Comment (whites11) [reply files/conf/k8s-addons:94]: good point

Comment (whites11) [reply files/conf/k8s-addons]: done

**Review by tfussell — APPROVED** [review]
LGTM.

**Comment by whites11** [comment]
> LGTM.

Thanks. Testing on Azure, and then merging

**Comment by whites11** [comment]
> > LGTM.
> 
> Thanks. Testing on Azure, and then merging

tests green

**Pull request merged**